File: app/src/main/java/com/example/dreamloaf/repository/AuthRepository.java
package com.example.dreamloaf.repository;

import android.app.Application;
import com.example.dreamloaf.data.AppDatabase;
import com.example.dreamloaf.data.User;
import com.example.dreamloaf.data.UserDao;
import java.util.concurrent.Executors;

public class AuthRepository {
    private final UserDao userDao;

    public AuthRepository(Application application) {
        AppDatabase db = AppDatabase.getInstance(application);
        userDao = db.userDao();
    }

    public boolean isLoginExists(String login) {
        return userDao.getUserByLogin(login) != null;
    }

    public void register(User user) {
        Executors.newSingleThreadExecutor().execute(() -> userDao.insert(user));
    }

    public User login(String login, String password) {
        return userDao.getUser(login, password);
    }
}

File: app/src/main/java/com/example/dreamloaf/repository/AuthRepository.kt
package com.example.dreamloaf.repository

import android.app.Application
import com.example.dreamloaf.data.AppDatabase
import com.example.dreamloaf.data.User
import com.example.dreamloaf.data.UserDao
import java.util.concurrent.Executors

class AuthRepository(application: Application?) {
    private val userDao: UserDao

    init {
        val db: AppDatabase = AppDatabase.getInstance(application!!.applicationContext)!!
        userDao = db.userDao()!!
    }

    fun isLoginExists(login: String?): Boolean {
        return userDao.getUserByLogin(login) != null
    }

    fun register(user: User) {
        Executors.newSingleThreadExecutor().execute { userDao.insert(user) }
    }

    fun login(login: String?, password: String?): User? {
        return userDao.getUser(login, password)
    }
}
